### PR TITLE
SWQ-28 game rules for all modes

### PR DESCRIPTION
**Pull request by axseinga**
-  Komponent zdefiniowany w odrębnym pliku .js
 - Komponent przyjmuje na wejściu napis do wyświetlenia.
 - Komponent wyświetla przekazany napis.
 -Komponent wyświetla nagłówek "Mode Rules".
 - Komponent wyświetla odpowiednią ikonę na lewo od nagłówka.

Instead of the component getting as argument full text I created a text inside it, the component takes game mode and right answer based on the current photo displayed and generates content. Version for all modes. 
Screenshot 2021-12-14 at 18 03 58

### CODE REVIEW

File: src/components/gameRules/gameRules.css [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.mode-rules_wrapper {
```
Comment (ossmate) [inline src/components/gameRules/gameRules.css]: why you mix dashes with underscores?

File: src/components/gameRules/gameRules.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import GAME_MODES from '../../consts/GAME_MODES';
+import './gameRules.css';
+
+const createTextContent = (gameMode, answer) => {
```
Comment (ossmate) [inline src/components/gameRules/gameRules.js]: what kind of text content it creates? 
function name indicates that it generate some message based on input - which it doesn't accept as parameter here

I think more meaningful name here could something like `gameDescriptionText`

File: src/components/gameRules/gameRules.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  return `You have one minute (1m) to answer as many questions as possible. During the game on each question you need to select ${
+    gameMode === GAME_MODES.people ? 'who' : 'what'
+  } from Star Wars is showed on the left (${answer} right now) from available options`;
+};
+
+const createGameRulesElement = (gameMode, answer) => {
```
Comment (ossmate) [inline src/components/gameRules/gameRules.js]: GameRulesContainer/Wrapper/Component

File: src/components/gameRules/gameRules.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  const rulesEl = document.createElement('div');
+  rulesEl.classList.add('mode-rules_content');
+
+  const content = createTextContent(gameMode, answer);
+  rulesEl.innerText = content;
+
+  containerEl.appendChild(titleEl);
+  containerEl.appendChild(rulesEl);
+
+  return containerEl;
```
Comment (ossmate) [inline src/components/gameRules/gameRules.js]: rulesContainer/Component

File: src/components/gameRules/gameRules.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+};
+
+export default createGameRulesElement;
```
Comment (ossmate) [inline src/components/gameRules/gameRules.js]: please avoid default exports

File: src/consts/GAME_MODES.js [hunk 6]
```diff
@@ -1,5 +1,7 @@
...
-    spaceships: 'spaceships',
-}
\ No newline at end of file
+const GAME_MODES = {
+  people: 'people',
+  vehicles: 'vehicles',
+  spaceships: 'spaceships'
+};
+
+export default GAME_MODES;
```
Comment (ossmate) [inline src/consts/GAME_MODES.js]: same as previous

**Review by ossmate — CHANGES_REQUESTED** [review]

File: src/components/gameRules/gameRules.js (hunk 5 above)
Comment (axseinga) [reply src/components/gameRules/gameRules.js]: Why avoid default exports?

File: src/components/gameRules/gameRules.css (hunk 1 above)
Comment (axseinga) [reply src/components/gameRules/gameRules.css]: Actually, I think it should be named "mode-rules". I used a dash to make it more readable. I got used to naming classes with BEM pattern. What name would you suggest in this case? Is camelCase good practice to name classes?

File: src/components/gameRules/gameRules.js (hunk 5 above)
Comment (ossmate) [reply src/components/gameRules/gameRules.js]: mainly because other developers across can import this to other module with different name
it cause potential problems with refactoring (e.g.: you are looking usage of createGameRuleElement to change it
's name (due to refactor) and in some place some developer imported it with differnt name - it can occur with potential error on production) 

tldr: its a good practice to avoid potential errors

**Review by ossmate — APPROVED** [review]

File: src/components/gameRules/gameRules.js (hunk 5 above)
Comment (axseinga) [reply src/components/gameRules/gameRules.js]: Thank you :)
